autotx/tests/integration/test_swap.py

```diff
@@ -1,121 +1,163 @@
-from autotx.utils.ethereum import (
-    get_erc20_balance,
-    get_native_balance,
-)
-from autotx.utils.ethereum.networks import NetworkInfo
+from decimal import Decimal
 from autotx.utils.ethereum.eth_address import ETHAddress
-from autotx.utils.ethereum.uniswap.swap import build_swap_transaction
-
-def test_swap(configuration):
-    (user, _, client, _) = configuration
-
-    network_info = NetworkInfo(client.w3.eth.chain_id)
-    weth_address = ETHAddress(network_info.tokens["weth"])
-    wbtc_address = ETHAddress(network_info.tokens["wbtc"])
+from autotx.utils.ethereum.lifi.swap import build_swap_transaction
+from autotx.utils.ethereum.networks import NetworkInfo
 
-    user_addr = ETHAddress(user.address)
+PLUS_DIFFERENCE_PERCENTAGE = 0.01
 
-    balance = get_erc20_balance(client.w3, wbtc_address, user_addr)
-    assert balance == 0
 
-    txs = build_swap_transaction(
-        client, 0.05, weth_address.hex, wbtc_address.hex, user_addr.hex, False
+def test_buy_one_usdc(configuration):
+    (_, _, client, manager) = configuration
+    network_info = NetworkInfo(client.w3.eth.chain_id)
+    eth_address = ETHAddress(network_info.tokens["eth"])
+    usdc_address = ETHAddress(network_info.tokens["usdc"])
+    desired_amount = 1
+    buy_usdc_with_eth_transaction = build_swap_transaction(
+        client,
+        desired_amount,
+        eth_address,
+        usdc_address,
+        manager.address,
+        False,
+        network_info.chain_id,
+    )
+    hash = manager.send_tx(buy_usdc_with_eth_transaction[0].tx)
+    manager.wait(hash)
+    usdc_balance = manager.balance_of(usdc_address)
+    assert usdc_balance >= desired_amount and usdc_balance <= desired_amount + (
+        desired_amount * PLUS_DIFFERENCE_PERCENTAGE
     )
 
-    for i, tx in enumerate(txs):
-        transaction = user.sign_transaction(
-            {
-                **tx.tx,
-                "nonce": client.w3.eth.get_transaction_count(user_addr.hex),
-                "gas": 200000,
-            }
-        )
-
-        hash = client.w3.eth.send_raw_transaction(transaction.rawTransaction)
-
-        receipt = client.w3.eth.wait_for_transaction_receipt(hash)
-
-        if receipt["status"] == 0:
-            print(f"Transaction #{i} failed ")
-            break
-
-    new_balance = get_erc20_balance(client.w3, wbtc_address, user_addr)
-    assert new_balance == 0.05
-
-def test_swap_recieve_native(configuration):
```

Comment:
Why do we not have this test anymore? I think it's a valid case to test to be able to get native

Reply:
i previously added this tests because we were manually handling the conversion from weth to eth - so i thought that it's okay to remove it, but will add it again